### ISSUE TITLE
Unable to save/copy after lengthy editing session

### ISSUE THREAD

**Issue by neuguy** [SabakiHQ/Sabaki#432]
On Linux, no engines.

I am creating/editing some sgf files, usually consisting of 10-20 nodes with comments. A few times now, after a lengthy editing session (about 2 hours) I cannot save the sgf. I can still edit the game tree in the viewer and all editing features appear to work within the viewer, but no additional changes can be saved and exiting without saving does not give the save prompt. Closing and reopening the file gives me what appears to be an earlier save, but with all variations pruned and only main line nodes remaining.

I have also tried opening a second instance of Sabaki and copying the nodes from the old instance to the new. This does not work, though it is possible to copy nodes from the new instance to the old one. Copying the sgf from the menu in the old instance does not add it to the clipboard.

I have been getting around this by saving and restarting Sabaki on relatively short intervals, and creating backups - I think saving and restarting every 30 minutes is probably enough for me to avoid these issues.

**Comment by yishn** [comment]
It would be helpful if you can provide some reproduction steps. What edits did you do?

**Comment by neuguy** [comment]
After some more testing, I have a more narrowed description of the problem and steps to reproduce.

![attempted_save](https://user-images.githubusercontent.com/42069788/43657496-e2ce2a5c-970a-11e8-91d0-89a03e8ad149.png)
![actual_save](https://user-images.githubusercontent.com/42069788/43657499-e4c4454e-970a-11e8-94da-63366610a6df.png)
[test1.sgf.zip](https://github.com/SabakiHQ/Sabaki/files/2258268/test1.sgf.zip)

Behavior:
After some undetermined number of edits are made to an initially blank sgf file, including the creation of a tree of variations, along with some undetermined number of saves, the sgf file stops saving accurately. Node that are not part of the line of main variations are pruned from any further saves. I did try adding a main variation node, which did save successfully, but this did not restore the pruned variations.

In the attached files:
attempted_save.png is a screenshot of an sgf file I created from a blank file from Sabaki following the steps to reproduce given below. Each node has a comment like the node in the screenshot, with the number x.y indicating the location in the tree for easier tracking when viewing the sgf in a text editor. I attempted to save this sgf file. The result is the attached sgf file, and actual_save.png is a screenshot of what you get when you open this sgf file.

Steps:

1. Open Sabaki, create a new sgf file, save as test1.sgf.
2. Toggle edit mode, create the board position in the screenshots. Save with ctrl-S after every stone placed.
3. Still in edit mode, create a comment for the node. I used !!!!!! TEST NODE x.y !!!!!! where x,y are numbers indicating tree height and variation number at each height. Save.
4. Copy the variation, then paste some number of times to create the tree in actual_save.png, saving each time. Edit the number x.y in each node to keep track later.
5. After saving for the final time, open test1.sgf in a new window or text editor without closing the original window, and compare.

Other details:
Ubuntu 16.04.2 LTS, Sabaki v0.35.0
